src/node/timer.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/logger.h"
+#include "ds/spinlock.h"
+
+#include <chrono>
+#include <functional>
+#include <list>
+
+namespace ccf
+{
+  using TimerCallback = std::function<bool()>;
+
+  /**
+   * A timer class to trigger actions periodically.
+   *
+   * A shared_ptr to a new Timer is returned by Timers::new_timer(period, cb).
+   * The timer is only active when it is started (start()) and marked as expired
+   * when the callback is triggered. If the callback returns true, the timer
+   * continues ticking. Otherwise, the timer expires and will tick again only
```

Comment:
Ah I misunderstood this. I think that if the callback returns true, the timer _restarts_?

Reply:
Exactly. If you want a timer that triggers forever, always return true from the callback. If you want it to stop calling after a while (like it is the case for the join RPC), return false when an arbitrary condition is met.